lib/ecmascript.ts

```diff
@@ -1532,107 +1527,94 @@ export const ES = ObjectAssign({}, ES2020, {
     return result;
   },
   CalendarMergeFields: (calendar, fields, additionalFields) => {
-    const mergeFields = ES.GetMethod(calendar, 'mergeFields');
-    if (mergeFields === undefined) return { ...fields, ...additionalFields };
-    const result = ES.Call(mergeFields, calendar, [fields, additionalFields]);
+    if (!calendar.mergeFields) {
+      return { ...fields, ...additionalFields };
+    }
+    const result = calendar.mergeFields(fields, additionalFields);
     if (ES.Type(result) !== 'Object') throw new TypeError('bad return from calendar.mergeFields()');
     return result;
   },
   CalendarDateAdd: (calendar, date, duration, options, dateAdd) => {
     if (dateAdd === undefined) {
-      dateAdd = ES.GetMethod(calendar, 'dateAdd');
+      dateAdd = calendar.dateAdd;
     }
-    const result = ES.Call(dateAdd, calendar, [date, duration, options]);
+    const result = calendar.dateAdd(date, duration, options);
```

Comment:
This is not quite correct, because if `dateAdd` is undefined then `calendar.dateAdd` will be accessed twice, potentially calling a getter twice. We have to do the equivalent of `dateAdd.call(calendar, date, duration, options)` but also being robust in the case that user code does `delete Function.prototype.call` which probably means saving `Reflect.apply` in a variable at the top of the file and using that to call the function with `calendar` as the this-value.

Reply:
Gotcha. I think the same applies to mergeFields?